stage1/init/common/pod.go

```diff
@@ -366,6 +366,60 @@ func findBinPath(p *stage1commontypes.Pod, appName types.ACName, app types.App,
 	return binPath, nil
 }
 
+// generateDeviceAllows generates a DeviceAllow= line for an app.
+// To make it work, the path needs to start with "/dev" but the device won't
+// exist inside the container. So for a given mount, if the volume is a device
+// node, we create a symlink to its target in "/rkt/volumes". Later,
+// prepare-app will copy those to "/dev/.rkt/" so that's what we use in the
+// DeviceAllow= line.
+func generateDeviceAllows(root string, appName types.ACName, mountPoints []types.MountPoint, mounts []mountWrapper, vols map[types.ACName]types.Volume, uidRange *user.UidRange) ([]string, error) {
+	var devAllow []string
+	var toShift []string
+
+	rktVolumeLinksPath := filepath.Join(root, "rkt", "volumes")
+	if err := os.MkdirAll(rktVolumeLinksPath, 0600); err != nil {
+		return nil, err
+	}
+	toShift = append(toShift, rktVolumeLinksPath)
+
+	for _, m := range mounts {
+		v := vols[m.Volume]
+		if v.Kind == "host" {
+			if fileutil.IsDeviceNode(v.Source) {
+				mode := "r"
+				if !IsMountReadOnly(v, mountPoints) {
+					mode += "w"
+				}
+
+				tgt := filepath.Join(common.RelAppRootfsPath(appName), m.Path)
+				// the DeviceAllow= line needs the link path in /dev/.rkt/
+				linkRel := filepath.Join("/dev/.rkt", v.Name.String())
+				// the real link should be in /rkt/volumes for now
+				link := filepath.Join(rktVolumeLinksPath, v.Name.String())
+
+				if _, err := os.Lstat(link); os.IsNotExist(err) {
+					if err := os.Symlink(tgt, link); err != nil {
```

Comment:
IIRC symlink() doesn't overwrite existing linkpaths. Any reason why you are doing lstat+symlink? In this context it may be fine, but in general it could open for races.